netbout-client/src/main/java/com/netbout/client/RtInbox.java

```diff
@@ -126,7 +126,9 @@ public Bout bout(final long number) {
     public Iterable<Bout> search(final String term) throws IOException {
         final List<Bout> result = new ArrayList<>(16);
         for (final Bout bout : this.iterate()) {
-            if (bout.messages().search(term).iterator().hasNext()) {
+            if (bout.title().contains(term)) {
+                result.add(bout);
+            } else if (bout.messages().search(term).iterator().hasNext()) {
```

Comment:
@krzyk that's right, we just need to know if the bout is elegible for the searching criteria. It doesn't matter whether it is because of title or message. That's why it would be an unnecessary extra processing to check the message when it is already eligible because of the title.
Even more, without the else, if the `term` were both in title and messages, it would be added twice to the `result` List